test/integration/functional-test.js

```diff
@@ -192,13 +198,36 @@ describe("add-on UI", () => {
       // Verify visible info in the item list.
       const listItemSubtitle = await waitFor(selectors.listItemSelectedSubtitle);
       expect(await listItemSubtitle.getText()).to.equal(expected.username);
+    });
+
+    const commonItemDelete = async () => {
+      const expected = {
+        origin: "https://foo.example.com",
+        username: "testUser",
+        password: "testPassword",
+      };
+      await addItem(expected);
+
+      // Verify visible info in the item list.
+      const listItem = await waitFor(selectors.listItemContainer);
+      listItem.click();
+
+      const deleteItemButton = await waitFor(selectors.deleteItemButton);
+      deleteItemButton.click();
+    };
+
+    it("can delete an existing item", async () => {
+      await commonItemDelete();
+      const confirmButton = await waitFor(selectors.confirmDialogConfirmButton);
+      confirmButton.click();
+      await waitUntilMissing(selectors.listItemContainer);
+    });
 
-      // Finally, ensure that our added item is found in Firefox
```

Comment:
As requested in standup, I just went ahead and deleted this code that checked for the login data in the API. Not really necessary, since the test already looks for the visible evidence.